File: frontend/bagshop/src/service/CartService.js
import axiosClient from "./AxiosConfig";

const CartService = {
  getAll() {
    return axiosClient.get("cart");
  },
  getCartById(cartId) {
    return axiosClient.get(`cart/${cartId}`);
  },
  getCartByAccountId(accountId) {
    return axiosClient.get(`cart/account/${accountId}`);
  },
  // Thêm sản phẩm vào giỏ hàng
  addToCart(accountId, cartRequest) {
    return axiosClient.post(`cart/add`, {
      accountId: accountId,
      items: cartRequest.items,
    });
  },
  // Tạo giỏ hàng mới
  createCart(cartRequest) {
    return axiosClient.post("cart", cartRequest);
  },
  updateCart(accountId, cartRequest) {
    return axiosClient.put(`cart/update`, {
      accountId: accountId,
      items: cartRequest.items,
    });
  },
  deleteCart(accountId, cartItemId) {
    return axiosClient.delete(`cart/${accountId}/delete/${cartItemId}`);
  },
};

export default CartService;

File: backend/src/main/java/bag/service/cart/CartService.java
package bag.service.cart;

import bag.modal.dto.CartDto;
import bag.modal.request.CartRequest;

import java.util.List;

public interface CartService {
    List<CartDto> getAllCarts();

    CartDto getCartById(int id);

    CartDto getCartByAccountId(int accountId);

    CartDto addToCart(CartRequest request);

    CartDto updateCartItem(CartRequest request);

    void deleteCartItem(int accountId, int cartItemId);



}
